dnf/rpm/miscutils.py

```diff
@@ -16,44 +16,84 @@
 from __future__ import print_function, absolute_import
 from __future__ import unicode_literals
 
-import rpm
+import re
 import os
 import subprocess
 import logging
-
-from dnf.i18n import ucd
-from dnf.i18n import _
 from shutil import which
 
+from dnf.i18n import _
 
-logger = logging.getLogger('dnf')
-
+_logger = logging.getLogger('dnf')
+_rpmkeys_binary = which("rpmkeys")
+_logger.debug(_('Using rpmkeys executable from {path} to verify signatures',
+              path=_rpmkeys_binary))
 
-def _verifyPkgUsingRpmkeys(package, installroot, fdno):
-    os.lseek(fdno, 0, os.SEEK_SET)
-    rpmkeys_binary = '/usr/bin/rpmkeys'
-    if not os.path.isfile(rpmkeys_binary):
-        rpmkeys_binary = which("rpmkeys")
-        logger.info(_('Using rpmkeys executable from {path} to verify signature for package: {package}.').format(
-            path=rpmkeys_binary, package=package))
+_sig_re = re.compile(rb'\AV[34] (?:RSA|DSA|EdDSA|ECDSA)/SHA(?:256|384|512) Signature, key ID [0-9a-f]{8}')
+_payload_digest_re = re.compile(rb'\APayload SHA(?:256|384|512) digest\Z')
 
-    if not os.path.isfile(rpmkeys_binary):
-        logger.critical(_('Cannot find rpmkeys executable to verify signatures.'))
+def _process_rpm_output(data):
+    if len(data) < 3 or data[0] != b'-:' or data[-1]:
+        return 2
+    seen_sig, missing_key, not_trusted, not_signed = False, False, False, False
+    valid = 0
+    for i in data[1:-1]:
+        if not i.startswith(b'    '):
```

Comment:
To be honest I don't really like this parsing in this section with the exact hardcoded indexes, using the regexes for the keys and such.

What about to trust the return code of `rpmkeys` and here just check if the lines end with: `NOKEY`/`NOTTRUSTED`/`NOTFOUND`/`OK`/`BAD` for the right return value?